pkg/parse/run.go

```diff
@@ -120,6 +126,24 @@ func Run(ctx context.Context, p Parser) {
 			// state of backoff retry.
 			statusUpdateTimer.Reset(opts.statusUpdatePeriod) // Schedule status update attempt
 
+		// Execute the entire parse-apply-watch loop for a namespace event.
+		case <-nsEventTimer.C:
```

Comment:
This case should probably be moved to the bottom of the select because it's the shortest timer. Technically, selects don't work top down, but people read them like switch statements, expecting the higher cases to take precedence. So since this is the shortest timer period, it should go last and let the others take precedence, at least for readability.

Reply:
FWIW, when multiple select channels have messages available, one of them in randomly selected to execute.

Reply:
^^ This is another reason this kind of timer based select is not optimal. Ideally these would all be re-written into a reconcile loop with an explicit precedence top to bottom, checking now against timestamps to see if enough time has elapsed to execute one of the cases.

Reply:
Moved to the bottom